### PR TITLE
Storage src

### LINKED ISSUE #3248: Extend build from source to STORAGE

### PR DESCRIPTION
**Pull request by gearama**
closes #3248 

Add ability to download source code for dependencies at build time for storage sdks. 
added ability to download the latest version of the dependencies from main. 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by vhvb1989** [comment]
Someone won't like the name of this PR 😄

**Review by vhvb1989 — APPROVED** [review]
Please wait for @Jinming-Hu approval before merging.

File: cmake-modules/FolderList.cmake
Comment (Jinming-Hu) [inline cmake-modules/FolderList.cmake:29]: Why do we need this PR? What problem are we trying to fix?

If we have cyclic dependencies(A depends on B and B depends on A), can everything still work?

Now there are two places we have to update when publishing a new version. Can we update this version with a pipeline?

Comment (gearama) [reply cmake-modules/FolderList.cmake:29]: we have customers requiring to build from source not using vcpkg. this PR fetches the sources from github for dependencies.

It should, i haven't seen any cyclic ones yet but all builds passed currently. 

since this feature is in development, i was actually thinking of parsing the vcpkg files to get the dependencies. since the versions of the deps don't change that often i left that feature towards the end.

**Comment by vhvb1989** [comment]
Hold merging this after Azure-Core release.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**
